Merging changes from GraalPy master and updating docs (#34)
graalvm/graalpy.github.io: docs/graalpy-embedded/reference/interop/other_langs_from_python.md

## Interacting with other dynamic languages from Python scripts

> More general, non-JVM specfic interactions with other languages from Python scripts are done ia the _polyglot_ API.

Reviewer: ```suggestion
More general, non-JVM specific interactions with other languages from Python scripts are achieved via the _polyglot_ API.
```

This includes all interactions with dynamic languages supported on GraalVM via the [Truffle framework](https://www.graalvm.org/latest/graalvm-as-a-platform/language-implementation-framework/), including JavaScript and Ruby.

### Installing other dynamic languages

> Other languages can be included in using their respective Maven dependencies in the same manner as GraalPy.

Reviewer: ```suggestion
Other languages can be included by using their respective Maven dependencies in the same manner as GraalPy.
```

> For example, given you have a Maven project with GraalPy already configured, add the following dependency to gain access to JavaScript as well:

Reviewer: ```suggestion
For example, if you have already configured a Maven project with GraalPy, add the following dependency to gain access to JavaScript:
```

```xml
<dependency>
    <groupId>org.graalvm.polyglot</groupId>
    <artifactId>js</artifactId>
    <version>23.1.1</version>
</dependency>
```

<aside markdown="1">

> For Python developers, other languages are only available for the GraalPy JVM distributions after using `libexec/graalpy-polyglot-get` from the distribution's root directory.

Reviewer: ```suggestion
For Python developers, other languages are only available for the GraalPy JVM distributions after using the `libexec/graalpy-polyglot-get` command from the distribution's root directory.
```

To install JavaScript, for example:
```shell
libexec/graalpy-polyglot-get js
```
</aside>

### Examples

1. Import the `polyglot` module to interact with other languages:
   ```python
   import polyglot
   ```

2. Evaluate inlined code in another language:
   ```python
   assert polyglot.eval(string="1 + 1", language="js") == 2
   ```

3. Evaluate code from a file:
   ```python
   with open("./my_js_file.js", "w") as f:
       f.write("Polyglot.export('JSMath', Math)")
   polyglot.eval(path="./my_js_file.js", language="js")
   ```

4. Import a glocal value from the polyglot scope:
   ```python
   Math = polyglot.import_value("JSMath")
   ```

    This global value should then work as expected:
    * Accessing attributes reads from the *polyglot members* namespace:
      ```python
      assert Math.E == 2.718281828459045
      ```

>     * Calling methods on the result tries to do a straight invoke and falls back to reading the member and trying to execute it.

Reviewer: ```suggestion
    * Calling a method on the result attempts to do a straight `invoke` and falls back to reading the member and trying to execute it.
```

      ```python
      assert Math.toString() == "[object Math]"
      ```

    * Accessing items is supported both with strings and numbers.
      ```python
      assert Math["PI"] == 3.141592653589793
      ```

5. Use the JavaScript regular expression engine to match Python strings:
   ```python
   js_re = polyglot.eval(string="RegExp()", language="js")

>     pattern = js_re.compile(".*(?:we have (?:a )?matching strings?(?:[!\\?] )?)(.*)")

Reviewer: these 4 lines are indented too far.
